templates/geosafe/analysis/dynamic_scripts/create/transparency_control.html

```diff
@@ -152,6 +152,31 @@
                 $("#impact-legend-img").attr("src", legend_url);
                 impact_layer.setZIndex(3);
             }
+
+            // Zooming magic
+            // If only one of the 3 layers is selected then zoom to the layer
+            if ((is_hazard != is_exposure) != is_impact) {
+              map.fitBounds([[layer_bbox_y1, layer_bbox_x1], [layer_bbox_y0, layer_bbox_x0]]);
+            }
+            // If hazard and exposure selected intersect both extents
+            if ((layer_purpose=='hazard' && is_exposure) || (layer_purpose=='exposure' && is_hazard)) {
+              // Intersect current extent to the input layer's extent
+              var new_sw = [
+                  Math.max(
+                      current_bounds.getSouthWest().lat,
+                      layer_bounds.getSouthWest().lat),
+                  Math.max(
+                      current_bounds.getSouthWest().lng,
+                      layer_bounds.getSouthWest().lng)];
+              var new_ne = [
+                  Math.min(
+                      current_bounds.getNorthEast().lat,
+                      layer_bounds.getNorthEast().lat),
+                  Math.min(
+                      current_bounds.getNorthEast().lng,
+                      layer_bounds.getNorthEast().lng)];
+              map.fitBounds([new_sw, new_ne]);
+            }
```

Comment:
Yes, I had a line (leaflet has a function to check if two bounds interersect) for that but deleted it since like you said, the exposure layers are filtered on the extent of hazard (or vice versa).

For the international dateline, let me check next week, as far as I know people would really leave it as #TODO in the code :)